tests/conftest.py

```diff
@@ -0,0 +1,149 @@
+import configparser
+import os
+from datetime import datetime, timedelta
+
+import pytest
+from fastapi.testclient import TestClient
+from sqlalchemy import create_engine, text
+from sqlalchemy.orm import sessionmaker, Session
+from ml_warehouse.schema import (
+    Base,
+    PacBioRun,
+    PacBioRunWellMetrics,
+    PacBioProductMetrics,
+    Study,
+    Sample,
+)
+from lang_qc.main import app
+from lang_qc.db.mlwh_connection import get_mlwh_db
+
+test_ini = os.path.join(os.path.dirname(__file__), "testdb.ini")
+
+
+@pytest.fixture(scope="session")
+def config() -> configparser.ConfigParser:
+    # Database credentials for the test MySQL instance are stored here. This
+    # should be an instance in a container, discarded after each test run.
+    test_config = configparser.ConfigParser()
+    test_config.read(test_ini)
+    yield test_config
+
+
+def mysql_url(config: configparser.ConfigParser):
+    """Returns a MySQL URL configured through an ini file.
+
+    The keys and values are:
+
+    [MySQL]
+    user       = <database user, defaults to "mlwh">
+    password   = <database password, defaults to empty i.e. "">
+    ip_address = <database IP address, defaults to "127.0.0.1">
+    port       = <database port, defaults to 3306>
+    schema     = <database schema, defaults to "mlwh">
+    """
+    section = "MySQL"
+
+    if section not in config.sections():
+        raise configparser.Error(
+            "The {} configuration section is missing. "
+            "You need to fill this in before running "
+            "tests on a {} database".format(section, section)
+        )
+    connection_conf = config[section]
+    user = connection_conf.get("user", "mlwh")
+    password = connection_conf.get("password", "")
+    ip_address = connection_conf.get("ip_address", "127.0.0.1")
+    port = connection_conf.get("port", "3306")
+    schema = connection_conf.get("schema", "mlwh")
+
+    return (
+        f"mysql+pymysql://{user}:{password}@"
+        f"{ip_address}:{port}/{schema}?charset=utf8mb4"
+    )
+
+
+def create_test_data(session: Session):
```

Comment:
This probably belongs in a fixture file in its own right, rather than hidden in the general test setup.